File: AEDS2/tp1/lab_treino/AquecimentoIterativo.c
#include <stdio.h>
#include <stdbool.h>

int str_len(char *palavra){
    int length = 0;
	while(*(palavra+length)){ 
		length++;
	}
	return(length);
}

bool str_cmp(char palavra1[], char palavra2[]){
    bool resp = true;
    int length = str_len(palavra1);

    if(length == str_len(palavra2)){
        for(int i = 0; i < length; i++){
            if(palavra1[i] == palavra2[i]){
               resp = false;
               i = length;
            } 
        }
    }
    return (resp);
}

int isUpper(char palavra[]){
    int contador = 0;
    for(int i = 0; i < str_len(palavra); i++){
        if(palavra[i] >= 'A' && palavra[i] <= 'Z'){
            contador++;
        }
    }
    return(contador);
}

int main(){
    char palavra[500];
    int contador = 0;
    
    scanf(" %[^\n]", palavra);  
    
    while(str_cmp(palavra, "FIM") != 0){
        contador = isUpper(palavra);
        printf("%d\n", contador);
        
        scanf(" %[^\n]", palavra);
    }

    return 0;
}

File: AEDS2/tp1/lab_treino/AquecimentoIterativo.java
import java.util.Scanner;

public class AquecimentoIterativo {
    
    public static boolean isEquals(String palavra1, String palavra2){
        int n = palavra1.length();
        boolean resp = true;
       
        if(n == palavra2.length()){
            for(int i = 0; i < n; i++){
                if(palavra1.charAt(i) == palavra2.charAt(i)){
                    resp = false;
                    i = n;
                }
            }
        }
        return (resp);
    }
    
    public static int isUpper(String palavra){
        int contador = 0;
        for(int i = 0; i < palavra.length(); i++){
            if(palavra.charAt(i) >= 'A' && palavra.charAt(i) <= 'Z'){
                contador++;
            }
        }
        return(contador);
    }
    
    public static void main(String[] args){
        String palavra;
        Scanner sc = new Scanner(System.in);
        int contador = 0;
        
        palavra = sc.nextLine();

        while(isEquals(palavra,"FIM")){
            contador = isUpper(palavra);
            System.out.println(contador);
            palavra = sc.nextLine();
        }
    }
}